Client/Frontend/UserContent/UserScripts/WalletEthereumProvider.js

```diff
@@ -0,0 +1,89 @@
+// Copyright 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+(function() {
+
+function post(method, payload) {
+  return new Promise((resolve, reject) => {
+    webkit.messageHandlers.$<handler>.postMessage({
+      "securitytoken": "$<security_token>",
+      "method": method,
+      "args": JSON.stringify(payload)
+    })
+    .then(resolve, (errorJSON) => {
+      try {
+        reject(JSON.parse(errorJSON))
+      } catch(e) {
+        reject(errorJSON)
+      }
+    })
+  })
+}
+
+Object.defineProperty(window, 'ethereum', {
```

Comment:
Is it possible for us to reuse the object definition in brave-core here: https://github.com/brave/brave-core/blob/8385231a90a9d9cb20ac98b0d96cf58bcffec20e/components/brave_wallet/resources/ethereum_provider.js

Reply:
That is already used from brave-core & injected, but it serves a different purpose. This JS file is the iOS equivalent of https://github.com/brave/brave-core/blob/master/components/brave_wallet/renderer/js_ethereum_provider.cc

Reply:
Ok thanks for clarifying - I don't think this would be an issue then.